src/pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -668,9 +669,7 @@ def __getstate__(self, drop_value: bool = True) -> dict:
         items = {k: v.__getstate__(drop_value=drop_value) for k, v in self.items()}
         return {**d, "items": items}
 
-    def __setstate__(
-        self, state: dict, map_location: Optional[Union[str, torch.device]] = None, sync_fn: Optional[Callable] = None
-    ) -> None:
+    def __setstate__(self, state: dict, map_location: MAP_LOCATION = None, sync_fn: Optional[Callable] = None) -> None:
```

Comment:
why are we making these changes? map location here is not the same as the other one. 
See the annotation: `Optional[Union[str, torch.device]]` and the use below in `self.to`

Reply:
reverted to https://github.com/Lightning-AI/lightning/pull/13932/commits/0502a4e9fe2f1b229265de8ae82a129e704f65ee and synced with upstream in https://github.com/Lightning-AI/lightning/pull/13932/commits/82fc4528431fd3412e30782ba64bcc65d2adba38